tests/test_suites/ShortSystemTests/test_ec_truncate_atomicity.sh

```diff
@@ -0,0 +1,36 @@
+timeout_set '1 minute'
+
+CHUNKSERVERS=6 \
+	DISK_PER_CHUNKSERVER=1 \
+	MOUNTS=5 \
+	MASTER_CUSTOM_GOALS="1 ec42: \$ec(4,2)" \
+	MOUNT_EXTRA_CONFIG="sfscachemode=NEVER" \
+	USE_RAMDISK=YES \
+	setup_local_empty_saunafs info
+
+# Create a source file -- a valid file-generated file which consists of 200 kB of data
+source="$RAMDISK_DIR/source"
+FILE_SIZE=200K file-generate "$source"
+
+cd "${info[mount0]}"
+
+# Create a file which consists of 400 kB of random data
+file="file"
+touch "$file"
+head -c 400K /dev/urandom > "$file"
+
+# Run in parallel 200 dd processes, each copies different 1 kB of data from the source file
+# Distribute these processes between 3 first mountpoints (mount0, mount1, mount2)
+for i in {0..199}; do
+	( sleep 0.$((2 * i + 100)) && dd if="$source" of="${info[mount$((i%3))]}/$file" \
+			bs=1KiB skip=$i seek=$i count=1 conv=notrunc 2>/dev/null ) &
+done
+# In the meanwhile, shorten file from 400K to 200K, chopping 1 kB in each of 200 steps
+# Sometimes share mountpoint with dd (mount2), sometimes not (mount3, mount4)
+for i in {399..200}; do
+	truncate -s ${i}K "${info[mount$((2 + i % 3))]}/$file"
+done
+wait # Wait for all dd processes to finish
+
+# Now file should be equal to the source file. Let's validate it!
+MESSAGE="Testing ec(4,2) config" assert_success file-validate "$file"
```

Comment:
This tests is similar to test_xor_truncate_atomicity. In the original test, the first stripe was removed in this point to check if the system can read also from the parities. Please, consider doing the same.

Reply:
Done